src/libraries/System.Private.CoreLib/src/System/String.cs

```diff
@@ -598,7 +598,7 @@ internal static unsafe int wcslen(char* ptr)
         {
             // IndexOf processes memory in aligned chunks, and thus it won't crash even if it accesses memory beyond the null terminator.
             // This IndexOf behavior is an implementation detail of the runtime and callers outside System.Private.CoreLib must not depend on it.
-            int length = SpanHelpers.IndexOf(ref *ptr, '\0', int.MaxValue);
+            int length = SpanHelpers.IndexOfNullCharacter(ref *ptr);
             if (length < 0)
```

Comment:
I assume that this is the only places that calls `IndexOfNullCharacter`. Move this check into `IndexOfNullCharacter` method so that there is less to inline?